File: leetCode/fukun/001-100/lc_003.js
const lengthOfLongestSubstring = function (str) {
    let m = 0;
    let _arr = [];
    for (let i = 0; i < str.length; i++) {
        let idx = _arr.indexOf(str[i]);
        _arr.push(str[i]);
        if (idx >= 0) {
            _arr = _arr.splice(idx + 1);
        }
        m = Math.max(m, _arr.length);
    }
    return m;
};

File: leetCode/lwc/0001-0100/lc_003.java
package solution2;

import java.util.HashMap;

public class U3 {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> map = new HashMap<>();
        int maxLength = 0;
        int flag = 0,head = 0;
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            if(!map.containsKey(c)) flag++;
            else{
                maxLength = Math.max(flag,maxLength);
                head = map.get(c)>head?map.get(c):head;
                flag = i-head;
            }
            map.put(c,i);
        }
        return Math.max(flag,maxLength);
    }
}
